apps/trading/components/banner/market-successor-banner.tsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import { isBefore, formatDuration, intervalToDuration } from 'date-fns';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import {
+  calcCandleVolume,
+  marketProvider,
+  useCandles,
+} from '@vegaprotocol/markets';
+import {
+  ExternalLink,
+  Intent,
+  NotificationBanner,
+} from '@vegaprotocol/ui-toolkit';
+import {
+  addDecimalsFormatNumber,
+  getMarketExpiryDate,
+  isNumeric,
+} from '@vegaprotocol/utils';
+import { t } from '@vegaprotocol/i18n';
+import * as Types from '@vegaprotocol/types';
+
+const getExpiryDate = (tags: string[], close?: string): Date | null => {
+  const expiryDate = getMarketExpiryDate(tags);
+  return expiryDate || (close && new Date(close)) || null;
+};
+
+export const MarketSuccessorBanner = () => {
+  const { marketId } = useParams();
+  const { pathname } = useLocation();
+  const isMarketPage = pathname.match(/^\/markets\/(?!(all)).+/);
```

Comment:
We could avoid of rendering by using router somewhere above

```
let element = useRoutes([
    {
      path: "/markets/:marketId",
      element: () => {
        const { marketId } = useParams();
        return marketId  === 'all' ? null : <MarketSuccessorBanner marketId={marketId} />
      },      
  ]);

  return element;
```

Reply:
it's a little bit outside of scope of this issue